Observer/CustomerAdminUpdate.php

```diff
@@ -0,0 +1,56 @@
+<?php
+    
+namespace Metrilo\Analytics\Observer;
+
+use Magento\Framework\Event\Observer;
+use Magento\Framework\Event\ObserverInterface;
+
+class CustomerAdminUpdate implements ObserverInterface
+{
+    
+    /**
+     * @param \Metrilo\Analytics\Helper\Data                    $helper
+     * @param \Metrilo\Analytics\Helper\ApiClient               $apiClient
+     * @param \Magento\Customer\Api\CustomerRepositoryInterface $customerRepository
+     */
+    public function __construct(
+        \Metrilo\Analytics\Helper\Data $helper,
+        \Metrilo\Analytics\Helper\ApiClient $apiClient,
+        \Magento\Customer\Api\CustomerRepositoryInterface $customerRepository
+    ) {
+        $this->helper             = $helper;
+        $this->apiClient          = $apiClient;
+        $this->customerRepository = $customerRepository;
+    }
+    
+    public function execute(Observer $observer)
+    {
+        try {
+            $customer     = $observer->getEvent()->getCustomer();
+            $origCustomer = $this->customerRepository->getById($customer->getId());
+            $storeId      = $this->helper->getStoreId();
+            
+            if (!trim($customer->getEmail())) {
```

Comment:
you have copied this code in 4 new places 
https://en.wikipedia.org/wiki/Don%27t_repeat_yourself